pkg/kubelet/kubelet_network.go

```diff
@@ -121,7 +121,7 @@ func (kl *Kubelet) providerRequiresNetworkingConfiguration() bool {
 	// is used or whether we are using overlay networking. We should return
 	// true for cloud providers if they implement Routes() interface and
 	// we are not using overlay networking.
-	if kl.cloud == nil || kl.cloud.ProviderName() != "gce" || kl.flannelExperimentalOverlay {
+	if kl.cloud == nil || kl.cloud.ProviderName() != "gce" {
```

Comment:
@freehan I think we have the same problem for not just flannel. If I'm using the gce cloud provider with weave and I don't want to configure routes, kubelet will report not ready. There is no way of opting out of gce routes with the gce cloud provider. Is that an issue?